src/Request.php

```diff
@@ -678,6 +678,30 @@ function ( $data ) {
 	 * @throws \Exception
 	 */
 	public function execute_http() {
+		// Validate content-type header for POST requests
```

Comment:
I'd SOLID and move this into a `protected function is_valid_content_type_header(): bool`

Reply:
Thanks for the feedback! 🙏🏻 

Pushed up a refactor with better abstraction of logic into more single use methods